MusicDB/1 table.sql

```diff
@@ -0,0 +1,56 @@
+Use MusicDB
+go
+drop table if exists instrument
+go 
+create table dbo.instrument(
+InstrumentID int not null identity primary key,
+CustomerFirstName varchar(15) not null
+    constraint ck_Instrument_Customer_first_name_cannot_be_blank check (CustomerFirstName > ''), 
+CustomerLastName varchar(15) not null
+    constraint ck_Instrument_Customer_last_name_cannot_be_blank check (CustomerLastName > ''), 
+Instrument varchar(20) not null
+    constraint ck_Instrument_Instrument_cannot_be_blank check (Instrument > ''), 
+Category varchar(15) not null
+    constraint ck_Instrument_Category_cannot_be_blank check (Category > ''),
+ID as concat(substring(Instrument, 1, 3), (substring(Category, 1,3))) persisted, 
+MonthlyRentalFee dec (4,2) not null, 
+DateRented date not null
+    constraint ck_Instrument_Date_rented_cannot_be_a_future_date check (DateRented < getdate()), 
+Profit as
+case 
+when DateReturned is not null then datediff(month, DateRented, DateReturned) * MonthlyRentalFee
+else 0
+end,
+DateReturned date null,
+EnteredIntoSystem as getdate(),
+    constraint ck_Instrument_customers_can_rent_for_a_minimum_of_1_month_to_a_maximum_of_12_months
+    check (datediff(month, DateRented, DateReturned) between 1 and 11)
+)
```

Comment:
-- LL check to ensure that Date Returned is after Date Rented
and check that the same customer is not renting the same instrument on the same date

Reply:
-- LL first constraint is good second needs more work. It should be making sure that the customer and instrument category are unique as well

Reply:
When I make the following columns: (CustomerFirstName, CustomerLastName,InstrumentName,Category,DateRented) all unique, I can change just the first name and it will go in. That is not good because I cannot have the same instrument being rented on the same day. Can you help me?

Reply:
-- LL Is this a problem? Aren't there multiple of the same instrument/category. (That's why there's a code I think...) If so, I would think it's not an issue to have the same type of instrument rented on the same day as long as it's rented by a different customer.